apps/console/src/features/users/components/add-user-updated.tsx

```diff
@@ -0,0 +1,947 @@
+import { UserManagementConstants } from "../constants";
+
+/**
+ * Proptypes for the add user component.
+ */
+export interface AddUserProps {
+    initialValues: any;
+    triggerSubmit: boolean;
+    emailVerificationEnabled: boolean;
+    onSubmit: (values: BasicUserDetailsInterface) => void;
+    hiddenFields?: ("userStore" |"userName" | "firstName" | "lastName" | "password" | "email")[];
+    requestedPasswordOption?: "ask-password" | "create-password";
+    isFirstNameRequired?: boolean;
+    isLastNameRequired?: boolean;
+    isEmailRequired?: boolean;
+    validationConfig?: ValidationDataInterface[];
+    setUserSummaryEnabled: (toggle: boolean) => void;
+    setAskPasswordFromUser?: (toggle: boolean) => void;
+    setOfflineUser?: (toggle: boolean) => void;
+    setSelectedUserStore?: (selectedUserStore: string) => void;
+    isBasicDetailsLoading?: boolean;
+    setBasicDetailsLoading?: (toggle: boolean) => void;
+    isUserStoreError: boolean;
+    readWriteUserStoresList: DropdownItemProps[];
+}
+
+/**
+ * Add user basic component.
+ *
+ * TODO: Add localization support. (https://github.com/wso2-enterprise/asgardeo-product/issues/209)
+ *
+ * @returns ReactElement
+ */
+export const AddUserUpdated: React.FunctionComponent<AddUserProps> = (
+    props: AddUserProps): ReactElement => {
+
+    const {
+        initialValues,
+        triggerSubmit,
+        emailVerificationEnabled,
+        onSubmit,
+        hiddenFields,
+        requestedPasswordOption,
+        isFirstNameRequired,
+        isLastNameRequired,
+        isEmailRequired,
+        setUserSummaryEnabled,
+        setAskPasswordFromUser,
+        setSelectedUserStore,
+        setOfflineUser,
+        isBasicDetailsLoading,
+        setBasicDetailsLoading,
+        validationConfig,
+        isUserStoreError,
+        readWriteUserStoresList
+    } = props;
+
+    const [ passwordOption, setPasswordOption ] = useState("create-password");
+    const [ askPasswordOption ] = useState<string>("email");
+    const [ password, setPassword ] = useState<string>("");
+    const [ userStoreRegex, setUserStoreRegex ] = useState<string>("");
+    const [ passwordConfig, setPasswordConfig ] = useState<ValidationFormInterface>(undefined);
+    const [ usernameConfig, setUsernameConfig ] = useState<ValidationFormInterface>(undefined);
+    const [ isValidPassword, setIsValidPassword ] = useState<boolean>(true);
+    const [ randomPassword, setRandomPassword ] = useState<string>(undefined);
+    const [ userstore, setUserstore ] = useState<string>(PRIMARY_USERSTORE);
+    const [ isValidEmail, setIsValidEmail ] = useState<boolean>(false);
+    const [ isEmailFilled, setIsEmailFilled ] = useState<boolean>(false);
+
+    const formBottomRef: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+    const emailRef: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+
+    const { t } = useTranslation();
+
+    const [ , setRegExLoading ] = useState<boolean>(false);
```

Comment:
We can remove this, can't we?

Reply:
Fixed with fd5839362baf8e4e50397013a20cfa8b0b15fa4e